### PR TITLE
Live auctions tech debt 📱

### PR DESCRIPTION
**Pull request by ashfurrow**
- [x] Fixes infinite loop in socket communications.
- [x] Cleans up socket comms code for Swift 3 style guidelines.
- [x] Fixes encoding of websocket ping.
- [x] Smooth out some edge cases.
- [x] Remove extraneous comments/print statements.
- [x] Factor gesture-driven animation into its own class.
- [x] Test that class.
- [x] Changelog entry.

Fix for https://github.com/artsy/auctions/issues/191.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">CHANGELOG isn’t valid YAML</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by ashfurrow** [comment]
So I'm at a bit of an impasse. I'd like to have something really slick where the appearing/disappearing animation of the CTA view is done in lockstep with the animations to the next/previous lot, but it's... very difficult 😅 Ideally this CTA would be a supplementary view of the collection view that displays the lot images, but the collection view is _under_ the bid history / lot info, so we can't.

I'll work on it and see what I can do.

**Comment by ashfurrow** [comment]
Cool, so the initial implementation is finished:



I'm really pleased with how it works. I'll amend the todo list in the PR to reflect the work that still needs to be done.

**Comment by orta** [comment]
This looks great 👯

**Comment by orta** [comment]
does it handle showing the lot information for a current lot well?

**Comment by ashfurrow** [comment]
Thanks! The view manages its own subscriptions to the salesperson for current lot updates, so we should be okay 👍

**Comment by orta** [comment]
@ashfurrow any clue about these reds? I can't grok em

**Comment by ashfurrow** [comment]
Test failures were just the code 65 nonsense with some misleading compiler warnings sprinkled on top. I've finished work on this, it's ready for review.

**Comment by orta** [comment]
will have to wait on this till tomorrow, please please ping me on slack (gonna have a busy day, and don't want to forget.)

File: Artsy/View_Controllers/LiveAuctionCurrentLotCTAPositionManager.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        if currentLot.lotID == focusedLot.lotID {
+            bottomPositionConstraint.constant = hiddenConstraintConstant
+        } else {
+            bottomPositionConstraint.constant = visibleConstraintConstant
+        }
+    }
+}
+
+// Provides a linear interpolation between a and b based on factor. Grabed from Wikipedia.
```
Comment (orta) [inline Artsy/View_Controllers/LiveAuctionCurrentLotCTAPositionManager.swift]: `grabbed`, if you will

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift [hunk 2]
```diff
@@ -173,42 +165,27 @@ class LiveAuctionLotViewController: UIViewController {
         bidHistoryViewController = historyViewController
         metadataStack.add(historyViewController, toParent: self, withTopMargin: "10", sideMargin: sideMargin)
 
-        let screenWidthIsLarge = UIScreen.main.applicationFrame.width > 320
+        let screenWidthIsLarge = UIScreen.main.bounds.width > 320
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift]: any reason for this? wouldn't it make it not work with splitting the screen on iPad?

Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift]: `applicationFrame` has been deprecated in iOS 10, this was the recommended replacement. It should work but I'll double-check.

Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift]: Looks good, though I did notice a bug on the iPad, fixed [here](https://github.com/artsy/eigen/pull/2048/commits/7d11b6f9a29c7bdc1ee8cadd5c0b93ae7dca846e#diff-813b6d57cb16ece9305a2072dbc6018dL85).

**Comment by ashfurrow** [comment]
🍏

**Comment by orta** [comment]
Cool 👍 looks good
